packages/shared/tsconfig.json

```diff
@@ -5,5 +5,6 @@
         "rootDir": "./"
     },
     "include": ["*.ts", "src/*.ts", "global.d.ts"],
-    "exclude": ["test/*.ts", "stories/*.ts"]
+    "exclude": ["test/*.ts", "stories/*.ts"],
+    "references": [{ "path": "../base" }]
```

Comment:
Likely a red herring, but the only reason I can think of that `shared` wouldn't be available when `overlay` was built is that it somehow beat `base` and then had missing types.